Classes/Domain/Repository/ProductRepository.php

```diff
@@ -70,25 +70,25 @@ public function createDemandQueryBuilder(DemandInterface $demand): QueryBuilder
             )
             ->from('tx_pxaproductmanager_domain_model_product');
 
-        $this->fireDemandEvent('afterDemandQueryBuilderInitialize', $demand, $queryBuilder);
+        $this->fireDemandEvent($demand, $queryBuilder);
```

Comment:
The fireDemandEvent() method is called four times here. How do we differentiate the different stages? Would it be better to call it once?